### PR TITLE
Include Tags in GetSubscription and Deployment APIs

### PR DESCRIPTION
**Pull request by ramoka178**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/resources/resource-manager/Microsoft.Resources/stable/2019-10-01/resources.json [hunk 1]
```diff
@@ -5068,6 +5089,7 @@
           "description": "The list of tag values."
         }
       },
+      "x-ms-azure-resource": true,
```
Comment (Tiano2017) [inline specification/resources/resource-manager/Microsoft.Resources/stable/2019-10-01/resources.json]: >"x-ms-azure-resource": true, [](start = 6, length = 28)

why do we need this?

**Review by hamidtirmizi — APPROVED** [review]

File: specification/resources/resource-manager/Microsoft.Resources/stable/2019-10-01/resources.json (hunk 1 above)
Comment (ramoka178) [reply specification/resources/resource-manager/Microsoft.Resources/stable/2019-10-01/resources.json]: Running autorest validation
"autorest --azure-validator --input-file=resources.json --openapi-type=arm" gave out few errors. Tried to resolve as much as possible. 

Error related to the above change was 
ERROR (XmsResourceInPutResponse/R2062/ARMViolation): The 200 response model for an ARM PUT operation must have x-ms-azure-resource extension set to true in its hierarchy. Operation: 'Tags_CreateOrUpdate' Model: 'TagDetails'.

If this is not mandatory, I could skip this.

Comment (hamidtirmizi) [reply specification/resources/resource-manager/Microsoft.Resources/stable/2019-10-01/resources.json]: Validation complains if this is missing for an ARM response

Comment (Tiano2017) [reply specification/resources/resource-manager/Microsoft.Resources/stable/2019-10-01/resources.json]: but what does this imply? Is TagDetails an "ARM resource"?

Comment (ramoka178) [reply specification/resources/resource-manager/Microsoft.Resources/stable/2019-10-01/resources.json]: TagDetails is in this format as well 
https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-azure-resource
The same has been marked for TagValue and TagsResource

Comment (hamidtirmizi) [reply specification/resources/resource-manager/Microsoft.Resources/stable/2019-10-01/resources.json]: `TagDetails` gets returned by one of our APIs `"/subscriptions/{subscriptionId}/tagNames/{tagName}"`. The validation complains, and this is the right fix for it. However, this is not related to the rest of this PR.

Comment (hamidtirmizi) [reply specification/resources/resource-manager/Microsoft.Resources/stable/2019-10-01/resources.json]: It is not necessarily an ARM resource. Somewhere in our specs we have said that anything ARM returns as a response to an API call, should have this.

**Review by Tiano2017 — APPROVED** [review]

File: specification/resources/resource-manager/Microsoft.Resources/stable/2019-10-01/resources.json (hunk 1 above)
Comment (Tiano2017) [reply specification/resources/resource-manager/Microsoft.Resources/stable/2019-10-01/resources.json]: I see. thanks for the details. I've approved the PR.

**Pull request merged**